function/run.go

```diff
@@ -19,9 +15,9 @@ type RunRequest struct {
 // RunResponse represents a response to a RunRequest. An instance of this
 // response struct is supplied as an argument to the Function type Run method.
 type RunResponse struct {
-	// Diagnostics report errors or warnings related to defining the function.
-	// An empty slice indicates success, with no warnings or errors generated.
-	Diagnostics diag.Diagnostics
+	// Error contains errors related to running the function.
+	// A nil error indicates success, with no errors generated.
+	Error *FuncError
```

Comment:
Maybe a shout here that `ConcatFuncErrors` can be utilized to simplify smushing together multiple errors? 🔨

Reply:
Have added a comment.